### DISCUSSION TITLE
Is running_state still a valid climate attribute?

### DISCUSSION THREAD

**Discussion by jkrzyszt** [Koenkk/zigbee2mqtt#25145] [Devices]
Hello,

I'm new to both Zigbee2Mqtt and Home Assistant.   I decided to refresh of my almost 20 years old home automation system based on X10 and Heyu.  I've already integrated it with HA, now I'm looking at new modern options.

I decided to switch from Heyu temperature sensors and Heyu controlled thermoelectric valves to Zigbee radiator thermostats for temperature control.  I already tried two models: Unitec 30946, recognized by Z2M as Siterwell GS361A-H04, and Schneider Electric CCTFR6100Z3, both listed among Z2M supported devices.  I planned to use their running_state attributes, which I expected to be translated by HA to hvac_action, as indicators of demand for heat that can control a boiler.  However, for both devices I tried Z2M reports "running_state":null.

Before I select another model to exercise if it supports running_state or still not, let me ask you if that attribute is still supported.  If it is then maybe I'm doing something wrong.  Any hints?

I'm running on a periodically updated Arch Linux, using distribution provided zigbee2mqtt package, now v1.42, and home-assistant v2024.11.3.  I'm using SLZB-06M as a Zigbee coordinator.

Thanks,
Janusz

**Comment by jkrzyszt** [comment]
Since CCTFR6100Z3 is recognized as WV704R0A0902, which means pi_heating_demand is available, I may be able to use that attribute, exposed by HA as a sensor, to control the boiler.  Then, running_state may be not needed in that case.

I'm still looking for something similar in Siterwell GS361A-H04.  I could calculate the demand for heat as (local_temperature < occupied_heating_setpoint) but temperature reports from that device look to me like coming not frequently enough and too late (which doesn't mean running_state would look better from that PoV if it worked as expected).

**Comment by jkrzyszt** [comment]
I've just received my first TS0601 (Tuya) thermostat and set it up.  So, answering myself: yes, running_state is still a valid climate attribute, only not every climate device variant with that attribute exposed supports it (correctly).

**Comment by MarkMikkelsen83** [comment]
I'm also getting null on running state
